ui/ducks/gas/gas.duck.js

```diff
@@ -78,138 +36,11 @@ export default function reducer(state = initState, action) {
-  return {
-    type: BASIC_GAS_ESTIMATE_STATUS,
-    value: status,
-  };
-}
-
-async function basicGasPriceQuery() {
-  const url = `https://api.metaswap.codefi.network/gasPrices`;
-  return await fetchWithCache(
-    url,
-    {
-      referrer: url,
-      referrerPolicy: 'no-referrer-when-downgrade',
-      method: 'GET',
-      mode: 'cors',
-    },
-    { cacheRefreshTime: 75000 },
-  );
-}
-
-export function fetchBasicGasEstimates() {
-  return async (dispatch, getState) => {
-    const isMainnet = getIsMainnet(getState());
-
-    dispatch(setBasicEstimateStatus(BASIC_ESTIMATE_STATES.LOADING));
-    let basicEstimates;
-    try {
-      dispatch(setEstimateSource(GAS_SOURCE.ETHGASPRICE));
-      if (isMainnet || process.env.IN_TEST) {
-        try {
-          basicEstimates = await fetchExternalBasicGasEstimates();
-          dispatch(setEstimateSource(GAS_SOURCE.METASWAPS));
-        } catch (error) {
-          basicEstimates = await fetchEthGasPriceEstimates(getState());
-        }
-      } else {
-        basicEstimates = await fetchEthGasPriceEstimates(getState());
-      }
-      dispatch(setBasicGasEstimateData(basicEstimates));
-      dispatch(setBasicEstimateStatus(BASIC_ESTIMATE_STATES.READY));
-    } catch (error) {
-      dispatch(setBasicEstimateStatus(BASIC_ESTIMATE_STATES.FAILED));
-    }
-
-    return basicEstimates;
-  };
-}
-
-async function fetchExternalBasicGasEstimates() {
-  const {
-    SafeGasPrice,
-    ProposeGasPrice,
-    FastGasPrice,
-  } = await basicGasPriceQuery();
-
-  const [safeLow, average, fast] = [
-    SafeGasPrice,
-    ProposeGasPrice,
-    FastGasPrice,
-  ].map((price) => new BigNumber(price, 10).toNumber());
-
-  const basicEstimates = {
-    safeLow,
-    average,
-    fast,
-  };
-
-  return basicEstimates;
-}
-
-async function fetchEthGasPriceEstimates(state) {
-  const chainId = getCurrentChainId(state);
-  const [cachedTimeLastRetrieved, cachedBasicEstimates] = await Promise.all([
-    getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES_LAST_RETRIEVED`),
-    getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES`),
-  ]);
-  const timeLastRetrieved = cachedTimeLastRetrieved || 0;
-  if (cachedBasicEstimates && Date.now() - timeLastRetrieved < 75000) {
-    return cachedBasicEstimates;
-  }
-  const gasPrice = await global.eth.gasPrice();
-  const averageGasPriceInDecGWEI = getValueFromWeiHex({
-    value: gasPrice.toString(16),
-    numberOfDecimals: 4,
-    toDenomination: 'GWEI',
-  });
-  const basicEstimates = {
-    average: Number(averageGasPriceInDecGWEI),
-  };
-  const timeRetrieved = Date.now();
-
-  await Promise.all([
-    setStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES`, basicEstimates),
-    setStorageItem(
-      `${chainId}_BASIC_PRICE_ESTIMATES_LAST_RETRIEVED`,
-      timeRetrieved,
-    ),
-  ]);
-
-  return basicEstimates;
-}
-
-export function setCustomGasPriceForRetry(newPrice) {
-  return async (dispatch) => {
-    if (newPrice === '0x0') {
```

Comment:
This had a guard against trying to retry transaction with a `0x0` price. I guess that is unnecessary now because the price will always come from the controller?

Reply:
yeah, that's correct.